### PR TITLE
Support multiple notifications

### LINKED ISSUE #51: Support for multiple notifications
If a notification is shown, and another comes up, it overwrites the current notification. There should either be a window for the notification list, or the notifications can stack on top of each other so when one goes away (maybe by clicking on it?), the other is shown.

### PR DESCRIPTION
**Pull request by le717**
Fixes #51. Finally got around to implementing this. Nothing too special about it. Main issue was getting new notifications to stack nicely, but I got it worked out.

Also reindented a few more scripts/areas according to code style guidelines.

### CODE REVIEW

**Comment by BoatsAreRockable** [comment]
Thanks for implementing this! I'm currently away but hopefully I can find some time to review and test this out in the next few days.

**Comment by le717** [comment]
I hope you like it. As I said, it's not too different from the old system, but different enough that I can't simply push it to `master`. :)

File: app/js/file.js [hunk 1]
```diff
@@ -149,6 +149,7 @@ module.exports = {};
     });
   }
 
+
```
Comment (BoatsAreRockable) [inline app/js/file.js]: Nit: the extra line is unnecessary here.

File: app/js/newsfeed.js [hunk 2]
```diff
@@ -148,6 +148,7 @@ module.exports = {};
       });
   }
 
+
```
Comment (BoatsAreRockable) [inline app/js/newsfeed.js]: Extra line also seems unnecessary here (unless 2 lines before the exports was intentional?)

Comment (le717) [inline app/js/newsfeed.js]: > (unless 2 lines before the exports was intentional?)

Yes.

File: app/js/notification.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  /**
+   * Hide a notification after a set time.
+   * @see _make
+   *
+   * @param {Object} notify - A notification object.
+   * @returns {Object} The notification object given.
+   */
+  function _hideDelayed(notify) {
```
Comment (BoatsAreRockable) [inline app/js/notification.js]: I'd suggest that this function and `_show` are reordered to make things easier to follow.

Comment (le717) [inline app/js/notification.js]: I am not following your reasoning here...

(BTW, `_hideDelayed()` is deliberately named that way. There are reasons. :) )

Comment (BoatsAreRockable) [inline app/js/notification.js]: @le717 `_make` returns an object to `_show`, which returns an object to `_hideDelayed`.

My thinking is that the order that these functions appear in the code should reflect this order.

**Comment by BoatsAreRockable** [comment]
These are looking very good! The only visual improvement I would suggest is adding a fade out animation when the notifications are hidden.

File: app/js/notification.js (hunk 3 above)
Comment (le717) [inline app/js/notification.js]: OK, makes sense. Will reorder ASAP.

Comment (BoatsAreRockable) [inline app/js/notification.js]: @le717 If you still think the current order is better then I'm happy to leave it how it is. I won't lose sleep over this! ;)

Comment (BoatsAreRockable) [inline app/js/notification.js]: @le717 missed your reply, please ignore my previous comment!

**Comment by le717** [comment]
Method moved per review. I will look into the fade out. The notification suddenly going away is very deliberate in design. (it was complicated enough just to get it to slide in), but I'll see what I can do. You can always go ahead and merge this if you'd like.

**Pull request merged**